pkg/querier/blocks_store_queryable.go

```diff
@@ -358,20 +359,20 @@ func (q *blocksStoreQuerier) selectSorted(sp *storage.SelectHints, matchers ...*
 		resQueriedBlocks  = []ulid.ULID(nil)
 	)
 
-	for retry := 0; retry < maxFetchSeriesTries; retry++ {
-		// Find the set of store-gateway instances having the blocks. The exclude list passed is the
+	for attempt := 1; attempt <= maxFetchSeriesAttempts; attempt++ {
+		// Find the set of store-gateway instances having the blocks. The exclude parameter is the
 		// map of blocks queried so far, with the list of store-gateway addresses for each block.
 		clients, err := q.stores.GetClientsFor(remainingBlocks, attemptedBlocks)
 		if err != nil {
 			// If it's a retry and we get an error, it means there are no more store-gateways left
 			// from which running another attempt, so we're just stopping retrying.
-			if retry > 0 {
+			if attempt > 1 {
```

Comment:
There can be different errors returned by `GetClientsFor`. Should we at least log it, just in case we need to investigate it in the future?

Reply:
You're right. What's about now?

Reply:
Thanks!